### PR TITLE
Try: Fix regression with featured images in latest posts.

### PR DESCRIPTION
**Pull request by jasmussen**
## What?

The Latest Posts block has an option to float featured images left and right:
Screenshot 2022-04-27 at 22 20 31

In the editor, this works:

Screenshot 2022-04-27 at 22 20 20

On the frontend, as of #38657, it doesnt, because the float rule is missing:

Screenshot 2022-04-27 at 22 20 24

This PR restores the float rule, but just for the latest posts block. 

## Testing Instructions

* Create a number of published posts that all have featured images
* Use the Latest Posts block
* Set featured images to be shown, and aligned left
* Observe that the editor and frontend look the same.

### CODE REVIEW

**Comment by jasmussen** [comment]
Here's a mostly cosmetic companion PR: https://github.com/WordPress/gutenberg/pull/40663

**Comment by carolinan** [comment]
This PR solves the positioning visually, but solving it also introduces an issue that is already present in the editor.
When the image is linked and aligned to the right, the focus first moves to the image on the right, and then back to the linked post title on the left side and this is unexpected.

**Comment by jasmussen** [comment]
Indeed, the order is all wrong in that setup. However given that issue is already present in the editor, and this PR just fixes the regression where floats didn't work at all, should it not land as a fix to that regression?

**Comment by jasmussen** [comment]
To elaborate a bit, I'm fixing this as a regression due to a report where a theme relied on this float rule being present, which is missing as of #38657. It seems to fix the tab-order issue we'd need a separate PR and/or issue.

**Review by carolinan — APPROVED** [review]

**Comment by jasmussen** [comment]
Thank you. Do you know if the issue you reported is ticketed? Otherwise I'm happy to open one.

**Comment by jasmussen** [comment]
CC: @Mamaduka should we backport this one?

**Comment by Mamaduka** [comment]
Thanks for the ping, @jasmussen.

I would defer to @adamziel and @gziolo on this subject :)

**Comment by jasmussen** [comment]
Created a ticket here: #40686

**Comment by adamziel** [comment]
I'm adding a backport label for now so this comes up during the triage tomorrow.

**Comment by adamziel** [comment]
I cherry picked this change into `wp/6.0` branch to be included in WordPress 6.0 Beta 4 on Monday: d8cac5b7ed

**Comment by Nyiriland** [comment]
👋  I was the one who initially pointed this bug out to Joen, and also just noticed that left alignment is broken not only on the front end, but also in the template editor, but strangely _not_ the page editor.

**Front end & Template editor:** 
image

**Page editor:**
image

**Testing Instructions**

- Create a number of published posts that all have featured images
- Use the Latest Posts block
- Set featured images to be shown, and aligned left
- Observe that the page editor, template editor, and frontend look the same.

**Pull request merged**
